CRM/Utils/Cache/SqlGroup.php

```diff
@@ -89,22 +107,82 @@ public function __construct($config) {
   /**
    * @param string $key
    * @param mixed $value
+   * @param null|int|\DateInterval $ttl
+   * @return bool
    */
-  public function set($key, &$value) {
-    CRM_Core_BAO_Cache::setItem($value, $this->group, $key, $this->componentID);
-    $this->frontCache[$key] = $value;
+  public function set($key, $value, $ttl = NULL) {
+    CRM_Utils_Cache::assertValidKey($key);
+
+    //    CRM_Core_BAO_Cache::setItem($value, $this->group, $key, $this->componentID);
+    $lock = Civi::lockManager()->acquire("cache.{$this->group}_{$key}._null");
+    if (!$lock->isAcquired()) {
+      CRM_Core_Error::fatal();
```

Comment:
@eileenmcnaughton @totten should we throw exception here rather than fatal?

Reply:
@seamuslee001 Probably. Thinking out loud:

* The system is generally simpler/clearer with exceptions.
* If there's anyone using `SqlGroup` who's trying to catch/handle problems, they could be surprised by a switch from `fatal` to `exception`.
* OTOH, very few things try to handle fatals. (Because they're awkward to handle.) And when they are handled, that's by converting the fatal to an exception. (Not strictly so - but in practical terms, that's all I've ever seen or heard of.)
* Overall, the previous contract for `CRM_Utils_Cache_Interface` was unclear about how errors would be reported. PSR-16 pushes us to standardize (i.e. returning booleans and throwing particular exception-interfaces). 
* I suspect real-world error-handling behaviors in `CRM/Utils/Cache/*` have been pretty varied (depending on the particular driver/configuration, you might get exceptions or fatals or FALSE return values). Any problem that rises from "making the error handling more consistent" is likely to be in the category of "the problem existed before but manifested differently".
* We don't often get to highlight the cache-subsystem as an area of change. It may be net-gentler on downstream to rip-off this particular bandaid rather than kick it down the road.

So... yes, I'll update to use the exception instead.